Firestore/core/src/firebase/firestore/immutable/sorted_container.h

```diff
@@ -42,6 +42,23 @@ class SortedMapBase {
    */
   using size_type = uint32_t;
 
+  /**
+   * A sentinel return value that indicates not found. Functionally similar to
+   * std::string::npos.
+   */
+  static constexpr size_type npos = static_cast<size_type>(-1);
+};
+
+/**
+ * A base class for implementing sorted maps, containing types and constants
+ * that don't depend upon the template parameters to the main class.
+ *
+ * Note that this exists as a base class rather than as just a namespace in
+ * order to make it possible for users of the SortedMap classes to avoid needing
+ * to declare storage for each instantiation of the template.
+ */
+class SortedMapBase : public SortedContainer {
```

Comment:
Can SortedMapBase (and SortedMapBase::kFixedSize in the .cc file) stay in the impl namespace?